src/hooks/useApi.js

```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import { API_URL } from "../utils";
+
+export const useApi = () => {
+  const token = localStorage.getItem("token");
+  const callApi = async (method, endPoint, isProtected = false, data = {}) => {
+    if (isProtected) endPoint = `${API_URL}/${endPoint}`;
+    else {
```

Comment:
suggestion: I don't see any difference these if else doing, do you need it?

Reply:
Go ahead everything seems good and code is clean good file and folder structure